tests/test_x509.py

```diff
@@ -173,6 +174,24 @@ def test_revoked_cert_retrieval(self, backend):
         # Check that len() works for CRLs.
         assert len(crl) == 12
 
+    def test_revoked_cert_retrieval_retain_only_revoked(self, backend):
+        """
+        This test attempts to trigger the crash condition described in
+        https://github.com/pyca/cryptography/issues/2557
+        """
+        crl = _load_cert(
+            os.path.join("x509", "custom", "crl_all_reasons.pem"),
+            x509.load_pem_x509_crl,
+            backend
+        )
+        revoked = crl[11]
+        crl = "overwritten"
+        # force a gc collection to potentially X509_CRL_free if there are
+        # no references to the X509_CRL left.
+        gc.collect()
```

Comment:
Is this actually necessary? I feel like on CPython the refcounts work out so that it's unnecesary.